snorkel/contrib/deps/test/labeling/model/test_label_model.py

```diff
@@ -0,0 +1,242 @@
+        self.assertEqual(lm.c_tree.node[0]["end_index"], 18)
+        # Singleton clique 1
+        self.assertSetEqual(lm.c_tree.node[1]["members"], frozenset({1}))
+        self.assertEqual(lm.c_tree.node[1]["start_index"], 3)
+        self.assertEqual(lm.c_tree.node[1]["end_index"], 6)
+
+    def test_init_params(self):
+        L = np.array([[0, 1, 0], [0, -1, 0]])
+        label_model = self._set_up_model(L, class_balance=[0.6, 0.4])
+
+        # mu_init = P(\lf=y|Y=y) = clamp(P(\lf=y) * prec_i / P(Y=y), (0,1))
+        # mu_init[lf0, lf2 = 1 | Y = 1] = clamp(1.0 * 0.7 / 0.6) = 1.0 since P(lf = 1) = 1.0
+        # mu_init[lf1 | Y = 2] = clamp(0.5 * 0.7 / 0.4) = 0.875 since P(lf = 2) = 0.5
+        mu_init = label_model.mu_init.numpy()
+        true_mu_init = np.array(
+            [
+                [1.0, 0.0],
+                [0.0, 0.0],
+                [0.0, 0.0],
+                [0.0, 0.875],
+                [1.0, 0.0],
+                [0.0, 0.0],
+                [0.0, 0.0],
+                [0.0, 0.0],
+                [0.0, 0.0],
+                [0.0, 0.0],
+            ]
+        )
+        np.testing.assert_array_equal(mu_init, true_mu_init)
+
+        # mu_init = P(\lf=y|Y=y) = clamp(P(\lf=y) * prec_i / P(Y=y), (0,1))
+        # mu_init[lf0, lf2 = 1 | Y = 1] = clamp(1.0 * 0.7 / 0.6) = 1.0 since P(lf = 1) = 1.0
+        # mu_init[lf1 = 2 | Y = 2] = clamp(0.5 * 0.7 / 0.7) = 0.5 since P(lf = 2) = 0.5
+        label_model._set_class_balance(class_balance=[0.3, 0.7], Y_dev=None)
+        label_model._init_params()
+
+        mu_init = label_model.mu_init.numpy()
+        true_mu_init = np.array(
+            [
+                [1.0, 0.0],
+                [0.0, 0.0],
+                [0.0, 0.0],
+                [0.0, 0.5],
+                [1.0, 0.0],
+                [0.0, 0.0],
+                [0.0, 0.0],
+                [0.0, 0.0],
+                [0.0, 0.0],
+                [0.0, 0.0],
+            ]
+        )
+        np.testing.assert_array_equal(mu_init, true_mu_init)
+
+    def test_build_mask(self):
+        L = np.array([[0, 0, 0], [1, 1, 0], [1, 0, 1], [0, 1, 1], [0, 1, 0]])
+        label_model = self._set_up_model(L)
+
+        # block diagonal with 0s for dependent LFs
+        # without deps, k X k block of 0s down diagonal
+        true_mask = np.array(
+            [
+                [0, 0, 1, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 1, 1, 0, 0, 0, 0, 0, 0],
+                [1, 1, 0, 0, 1, 1, 1, 1, 1, 1],
+                [1, 1, 0, 0, 1, 1, 1, 1, 1, 1],
+                [0, 0, 1, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 1, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 1, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 1, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 1, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 1, 1, 0, 0, 0, 0, 0, 0],
+            ]
+        )
+
+        mask = label_model.mask.numpy()
+        np.testing.assert_array_equal(mask, true_mask)
+
+
+@pytest.mark.complex
```

Comment:
can we add:
* a test that explicitly checks the conversion from low rank to mu for the dependent case
* a test for multi-class (can be complex)
* a test comparing the `weights` explicitly to the recovered weights (like [here](https://github.com/snorkel-team/snorkel/blob/77e36c644fb2a93467514230c72fa776d9229ab5/test/labeling/model/test_label_model.py#L168))

Reply:
discussion IRL about checking conditional probabilities instead of weights

Reply:
Added a multiclass test.